src/compas/datastructures/mesh/_mesh.py

```diff
@@ -2620,6 +2643,78 @@ def face_flatness(self, fkey):
         face = range(len(self.face_vertices(fkey)))
         return flatness(vertices, [face])[0]
 
+    def face_aspect_ratio(self, fkey):
+        """Face aspect ratio as the ratio between the lengths of the maximum and minimum face edges.
+
+        Parameters
+        ----------
+        fkey : Key
+            The face key.
+
+        Returns
+        -------
+        float
+            The aspect ratio.
+          
+         References
+        ----------
+        .. [1] Wikipedia. *Types of mesh*.
+               Available at: https://en.wikipedia.org/wiki/Types_of_mesh.
+
+        """
+
+        face_edge_lengths = [self.edge_length(u, v) for u, v in self.face_halfedges(fkey)]
+        return max(face_edge_lengths) / min(face_edge_lengths)
+
+    def face_skewness(self, fkey):
+        """Face skewness as the maximum absolute angular deviation from the ideal polygon angle.
+
+        Parameters
+        ----------
+        fkey : Key
+            The face key.
+
+        Returns
+        -------
+        float
+            The skewness.
+          
+         References
+        ----------
+        .. [1] Wikipedia. *Types of mesh*.
+               Available at: https://en.wikipedia.org/wiki/Types_of_mesh.
+
+        """
+
+        ideal_angle = 180 * (1 - 2 / float(len(self.face_vertices(fkey))))
+        
+        angles = [angle_points(self.vertex_coordinates(v), self.vertex_coordinates(u), self.vertex_coordinates(w), deg = True) for u, v, w in window(self.face_vertices(fkey), n = 3, wrap = True)]
+
+        return max((max(angles) - ideal_angle) / (180 - ideal_angle), (ideal_angle - min(angles)) / ideal_angle)
+
+    def face_curvature(self, fkey):
+        """Dimensionless face curvature as the maximum face vertex deviation from the best-fit plane of the face vertices divided by the average lengths of the face vertices to the face centroid.
+
+        Parameters
+        ----------
+        fkey : Key
+            The face key.
+
+        Returns
+        -------
+        float
+            The dimensionless curvature.
+
+        """
+
+        plane = bestfit_plane([self.vertex_coordinates(vkey) for vkey in self.vertices()])
+
+        max_deviation = max([distance_point_plane(self.vertex_coordinates(vkey), plane) for vkey in self.vertices()])
+
+        average_distances = avrg([distance_point_point(self.vertex_coordinates(vkey), self.face_centroid(fkey)) for vkey in self.vertices()])
+
+        return max_deviation / average_distances
+
```

Comment:
this is more of a question than a comment, but should these not be in separate modules? they have little to do with the data structure itself... (i am aware that i have also not always been consistent in this)